src/main/java/uk/gov/hmcts/reform/et/syaapi/service/utils/CaseRoleManagementServiceUtil.java

```diff
@@ -0,0 +1,197 @@
+     */
+    public static String getRespondentFullNameByUserInfo(UserInfo userInfo) {
+        if (ObjectUtils.isEmpty(userInfo)) {
+            return StringUtils.EMPTY;
+        }
+        return ((StringUtils.isNotBlank(userInfo.getName()) ? userInfo.getName()
+            : StringUtils.isNotBlank(userInfo.getGivenName()) ? userInfo.getGivenName() : StringUtils.EMPTY)
+            + StringUtils.SPACE
+            + (StringUtils.isNotBlank(userInfo.getFamilyName()) ? userInfo.getFamilyName() : StringUtils.EMPTY)).trim();
+    }
+
+    /**
+     * Gets notification number by the notification collection in case data. If there is no notification collection
+     * returns 1 else returns number of elements in notification collection plus 1.
+     * @param caseData case data to check notification collection size.
+     * @return next available number to be used as notification id.
+     */
+    public static int getNextNotificationNumber(CaseData caseData) {
+        if (CollectionUtils.isEmpty(caseData.getSendNotificationCollection())) {
+            return 1;
+        }
+        return caseData.getSendNotificationCollection().size() + 1;
+    }
+
+    /**
+     * Generates HttpHeaders with the given user and service authorisation tokens.
+     * @param authToken authorisation token of the claimant
+     * @param serviceAuthorisation service authorisation created by authorisation token generator
+     * @return org.springframework.http.HttpsHeaders to call remote APIs
+     * @throws IOException Thrown exception when authorisation token does not match with the authorisation token regex
+     *                     which is [a-zA-Z0-9._\s\S]+$
+     */
+    public static HttpHeaders buildHeaders(String authToken, String serviceAuthorisation) throws IOException {
+        if (authToken.matches(AUTHORISATION_TOKEN_REGEX)) {
+            HttpHeaders headers = new HttpHeaders();
+            headers.add(HEADER_AUTHORIZATION, authToken);
+            headers.add(HEADER_SERVICE_AUTHORIZATION, serviceAuthorisation);
+            headers.add(HEADER_CONTENT_TYPE, HEADER_VALUE_APPLICATION_JSON);
+            return headers;
+        } else {
+            throw new IOException(EXCEPTION_AUTHORISATION_TOKEN_REGEX);
+        }
+    }
+
+    /**
+     * Generates case data with role modification notification by adding send notification type item to send
+     * notification collection of the CaseData model.
+     * @param userInfo user info generated by user token. Used to get notification subject list.
+     * @param caseDetails case details received by case service with case submission id.
+     * @param notifyUserCaseRoleModificationRequest details of the modification received from client. It has
+     *                                              caseSubmissionReference, role, modificationType, caseType
+     * @return CaseData that has modification notification info.
+     */
+    public static CaseData generateCaseDataByUserInfoCaseDetails(
```

Comment:
I don't think any of this is needed? We don't need to send out any notifications and especially not from the send notification bit

Reply:
if an email is to go out if a respondent has assigned themselves to a case, it would not be through the send notification stuff as that is initially sent from the tribunal

Reply:
please check with Dan on this

Reply:
Harpreet, this is for showing the notification in response-hub. It doesn't send any e-mail or something else. As in the image below. To manage showing the notification and not showing it I  will be using CaseData.SendNotificationCollection, instead of having a new Notification Collection for Respondents. 

I would be grateful, If there is a better field that I can use for showing notifications to respondents.

Reply:
The problem with using the sendNotification collection is the caseworker would then see an entry on ExUI like this
![Screenshot 2024-08-27 at 13 48 43](https://github.com/user-attachments/assets/7423de5b-ccdc-407b-a9c4-9e7dc959e706)

On et-sya, we do show notifications banners outside of the sendNotification as well.
- https://github.com/hmcts/et-sya-frontend/blob/master/src/main/controllers/citizen-hub/CitizenHubController.ts#L194-L197
- https://github.com/hmcts/et-sya-frontend/blob/master/src/main/controllers/helpers/CitizenHubHelper.ts#L32-L67
- https://github.com/hmcts/et-sya-frontend/blob/master/src/main/views/citizen-hub-notification-banner.njk

We'd probably want something like this where the banner constantly shows until a response has been received

Reply:
Removed all updates for Case Role Management notification.